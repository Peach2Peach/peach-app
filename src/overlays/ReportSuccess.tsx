import React, { ReactElement, useContext } from 'react'
import { View } from 'react-native'

import tw from '../styles/tailwind'

import { Headline, Icon, PrimaryButton, Text } from '../components'
import i18n from '../utils/i18n'

import { OverlayContext } from '../contexts/overlay'
import { account } from '../utils/account'
import { useNavigation } from '../hooks'

export default (): ReactElement => {
  const navigation = useNavigation()
  const [, updateOverlay] = useContext(OverlayContext)

  const closeOverlay = () => {
    updateOverlay({ visible: false })
  }

  const goToHome = () => {
    navigation.replace(account?.publicKey ? 'home' : 'welcome')
    closeOverlay()
  }
  return (
    <View style={tw`px-6`}>
      <Headline style={tw`text-3xl leading-3xl text-white-1`}>{i18n('report.success.title')}</Headline>
      <View style={tw`flex items-center mt-3`}>
        <View style={tw`flex items-center justify-center w-16 h-16 bg-green rounded-full`}>
          <Icon id="check" style={tw`w-12 h-12`} color={tw`text-white-1`.color} />
        </View>
      </View>
      <Text style={tw`text-center text-white-1 mt-5`}>{i18n('report.success.text.1')}</Text>
      <Text style={tw`text-center text-white-1 mt-5`}>{i18n('report.success.text.2')}</Text>
      <PrimaryButton style={tw`self-center mt-5`} onPress={goToHome} narrow>
        {i18n('report.success.backHome')}
      </PrimaryButton>
    </View>
  )
}
