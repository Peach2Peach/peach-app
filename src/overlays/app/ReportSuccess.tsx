import { View } from 'react-native'
import { PeachText } from '../../components/text/Text'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'

export const ReportSuccess = () => (
  <View>
    <PeachText style={tw`my-2`}>{i18n('report.success.text.1')}</PeachText>
    <PeachText>{i18n('report.success.text.2')}</PeachText>
  </View>
)
