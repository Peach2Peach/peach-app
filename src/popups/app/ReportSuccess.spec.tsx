import { createRenderer } from 'react-test-renderer/shallow'
import { ReportSuccess } from './ReportSuccess'

describe('ReportSuccess', () => {
  const shallowRenderer = createRenderer()
  it('should render correctly', () => {
    shallowRenderer.render(<ReportSuccess />)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })
})
