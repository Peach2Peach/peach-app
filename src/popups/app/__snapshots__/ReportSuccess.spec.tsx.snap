// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportSuccess should render correctly 1`] = `
<View>
  <PeachText
    style={
      {
        "marginBottom": 8,
        "marginTop": 8,
      }
    }
  >
    We'll normally be in touch within 24 hours on workdays.Â 
  </PeachText>
  <PeachText>
    Speak to you soon!
  </PeachText>
</View>
`;
